tokio-uds/src/datagram.rs

```diff
@@ -98,30 +99,32 @@ impl UnixDatagram {
     /// On success, returns the number of bytes read and the address from
     /// whence the data came.
     pub fn poll_recv_from(&self, buf: &mut [u8]) -> Poll<(usize, SocketAddr), io::Error> {
-        if self.io.poll_read_ready(Ready::readable())?.is_not_ready() {
-            return Ok(Async::NotReady);
+        try_ready!(self.io.poll_read_ready(Ready::readable()));
+
+        match self.io.get_ref().recv_from(buf) {
+            Ok(ret) => Ok(ret.into()),
+            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+                self.io.clear_read_ready(mio::Ready::readable())?;
+                Ok(Async::NotReady)
+            }
+            Err(e) => Err(e),
         }
-        let r = self.io.get_ref().recv_from(buf);
-        if is_wouldblock(&r) {
-            self.io.clear_read_ready(Ready::readable())?;
-            return Ok(Async::NotReady);
-        }
-        r.map(Async::Ready)
     }
 
     /// Receives data from the socket.
     ///
     /// On success, returns the number of bytes read.
     pub fn poll_recv(&self, buf: &mut [u8]) -> Poll<usize, io::Error> {
-        if self.io.poll_read_ready(Ready::readable())?.is_not_ready() {
-            return Ok(Async::NotReady);
-        }
-        let r = self.io.get_ref().recv(buf);
-        if is_wouldblock(&r) {
-            self.io.clear_read_ready(Ready::readable())?;
-            return Ok(Async::NotReady);
+        try_ready!(self.io.poll_read_ready(Ready::readable()));
+
+        match self.io.get_ref().recv(buf) {
+            Ok(ret) => Ok(ret.into()),
+            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+                self.io.clear_read_ready(mio::Ready::readable())?;
```

Comment:
Change this to `Ready::readable()` and drop the `use mio;` import.

Reply:
Done